enhancements/windows-containers/container-runtime-containerd.md

```diff
@@ -0,0 +1,184 @@
+    * register containerd as service
+    * start containerd service 
+
+## Network changes
+Current CNI/IPAM will be used for containerd and no change in HNS-Netwrok and HNS-Ednpoint creation
+steps. The config file which will be used to by containerd will point to same CNI/IPAM executables. 
+
+## feature flag
+Containerd feature flag will be used to enable this feature. This will be enabled by default in 
+community operator for customers and developers to try it out.  This will become default for openshift 
+4.11. At any given point in time we have one runtime support and do not allow switching between runtime.
+
+## logging
+Containerd can be started with parameters in which we can enable logging and specify file path to 
+log warnings/errors. Log files will be stored at c:\var\log\containerd
+
+## upgrade 
+When a cluster is upgraded OLM will switch to using a new Red Hat operators
+index. Because WMCO is named the same in both indexes, OLM will upgrade WMCO
+from the previous version, up to the latest version available in the new
+cluster.
+
+The procedure for an upgrade is as follows:
+1) As part of upgrade, basic validation will be done and de-configuring  will take place.
+2) During De-configuration, older version of kubelet, kube-proxy and CNI will be uninstalled.
+3) The newer version of WMCO will install all the required components along with containerd. 
+4) Kuebelet will start using containerd for pulling and managing container images. 
+
+   
+### Risks and Mitigations
+
+* If the cluster is upgraded, and the new version introduces an issue due to containerd
+  that should be addressed coz docker runtime support won't be available in kubelet.
+  going back to older WMCO might also run into issue due to kubelet version mismatch
+  between API server and kubelet.As we are planning to bring this feature before 4.11
+  this can be tested well and should make sure we address all the issues by working
+  with containerd open source community.
+* containerd doesn't support image-pull-progress-deadline as of now. There is a PR
+  https://github.com/containerd/containerd/pull/6150 work in progress. Until this 
+  gets merged, if windows image pull takes more time than the default value, we might
+  run into to image pull timeout error. proposed solution would be pull the image first
+  with ctr or crictl commandline tool and then create pods.
+* Currently windows_exporter has been used to collect metrics from windows node. we do 
+  have containerd support in https://github.com/prometheus-community/windows_exporter/releases/tag/v0.16.0
+  All the functionalities supported in docker runtime need to be checked for containerd.
+
+### Test Plan
+
+* We will have new e2e test case to be added for containerd replicating same existing 
+  test case that covers WMCO functionality. 
+* Containerd is agnostic to platform so testing in any platform should be fine.
+* Update WMCO community image on release repo so that CI workflow will use containerd
```

Comment:
This is not accurate. We will update the OKD e2e tests to use the containerd feature flag. The community image is built independent of CI in quay.